diff --git a/src/core/config/config.common.ts b/src/core/config/config.common.ts
index 7eb30d60584903f2515f2b88a67c804a363f11b9..7fce1f3405c8c389c399d48fdc9d76414bba19ad 100644
--- a/src/core/config/config.common.ts
+++ b/src/core/config/config.common.ts
@@ -4,6 +4,8 @@ import { UniwindListener } from '../listener'
 import { CSSVariables, GenerateStyleSheetsCallback, ThemeName } from '../types'
 
 const SYSTEM_THEME = 'system' as const
+const RN_VERSION = Platform.constants.reactNativeVersion.minor
+const UNSPECIFIED_THEME = RN_VERSION >= 82 ? 'unspecified' : undefined
 
 export class UniwindConfigBuilder {
     protected themes = ['light', 'dark']
@@ -12,7 +14,10 @@ export class UniwindConfigBuilder {
 
     constructor() {
         Appearance.addChangeListener(event => {
-            const colorScheme = event.colorScheme ?? ColorScheme.Light
+            // @ts-expect-error RN >0.82 - breaking change
+            const colorScheme = event.colorScheme === 'unspecified'
+                ? ColorScheme.Light
+                : event.colorScheme ?? ColorScheme.Light
             const prevTheme = this.#currentTheme
 
             if (this.#hasAdaptiveThemes && prevTheme !== colorScheme) {
@@ -32,7 +37,14 @@ export class UniwindConfigBuilder {
     }
 
     private get colorScheme() {
-        return Appearance.getColorScheme() ?? ColorScheme.Light
+        const colorScheme = Appearance.getColorScheme()
+
+        // @ts-expect-error RN >0.82 - breaking change
+        if (colorScheme === 'unspecified') {
+            return ColorScheme.Light
+        }
+
+        return colorScheme ?? ColorScheme.Light
     }
 
     // eslint-disable-next-line @typescript-eslint/no-redundant-type-constituents
@@ -47,7 +59,8 @@ export class UniwindConfigBuilder {
                 this.#currentTheme = this.colorScheme
 
                 if (Platform.OS !== 'web') {
-                    Appearance.setColorScheme(undefined)
+                    // @ts-expect-error RN >0.82 - breaking change
+                    Appearance.setColorScheme(UNSPECIFIED_THEME)
                 }
 
                 return
@@ -61,7 +74,12 @@ export class UniwindConfigBuilder {
             this.#currentTheme = theme
 
             if (Platform.OS !== 'web') {
-                Appearance.setColorScheme(isAdaptiveTheme ? this.#currentTheme as ColorScheme : undefined)
+                Appearance.setColorScheme(
+                    // @ts-expect-error RN >0.82 - breaking change
+                    isAdaptiveTheme
+                        ? this.#currentTheme as ColorScheme
+                        : UNSPECIFIED_THEME,
+                )
             }
         } finally {
             if (prevTheme !== this.#currentTheme) {
diff --git a/uniwind.css b/uniwind.css
index ccdf104d8d870b4c5d850d84d42be19767e87615..c11cec7f9b462a016d4ed6151a889ec91fc927e1 100644
--- a/uniwind.css
+++ b/uniwind.css
@@ -378,3 +378,25 @@
 
 @custom-variant light (&:where(.light, .light *));
 @custom-variant dark (&:where(.dark, .dark *));
+
+@theme {
+    --color-border: unset;
+    --color-input: unset;
+    --color-ring: unset;
+    --color-background: unset;
+    --color-foreground: unset;
+    --color-primary: unset;
+    --color-primary-foreground: unset;
+    --color-secondary: unset;
+    --color-secondary-foreground: unset;
+    --color-destructive: unset;
+    --color-destructive-foreground: unset;
+    --color-muted: unset;
+    --color-muted-foreground: unset;
+    --color-accent: unset;
+    --color-accent-foreground: unset;
+    --color-popover: unset;
+    --color-popover-foreground: unset;
+    --color-card: unset;
+    --color-card-foreground: unset;
+}
\ No newline at end of file
